rfcs/rfc-snafu-error-handling.md

```diff
@@ -0,0 +1,265 @@
+# RFC: Hierarchical Error Handling with Snafu
+
+## Summary
+
+This RFC proposes migrating Vortex's error handling from a centralized `VortexError` enum to a hierarchical error system using the [`snafu`] library. The migration would proceed incrementally, with each crate defining its own error types that convert into the existing `VortexError`, allowing for a gradual transition without breaking changes. Once the migration is done, we can remove the current `VortexError` and replace it with a top-level, user-facing `VortexError`.
+
+[`snafu`]: https://docs.rs/snafu/latest/snafu/
+
+## Motivation
+
+Vortex currently uses a single `VortexError` enum shared across the entire codebase. While this approach is relatively consistent and easy, it has several limitations that has clearly become increasingly problematic.
```

Comment:
I would strike " that has clearly become increasingly problematic.", it's a bit too subjective perhaps?